tests/unit/fem/test_bdr_edgedof.cpp

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "unit_tests.hpp"
+#include "mfem.hpp"
+#include "../mesh/mesh_test_utils.hpp"
+
+using namespace mfem;
+
+#ifdef MFEM_USE_MPI
+
+namespace boundary_edge_dof_test
+{
+
+// Generate all possible partitionings of n_elements into num_procs
```

Comment:
The current MFEM unit tests run quickly, and we should try to keep it that way. These unit tests are too slow in parallel, as the number of cases produced by using all possible partitions is extremely large, even for 2 MPI ranks. For example, 

srun -n 2 punit_tests "[BoundaryEdgeDoFs]"
All tests passed (131170 assertions in 3 test cases)

Can this test be reduced to a small number of cases that test the capability sufficiently?